### PR TITLE
[dagit telemetry] log telemetry via async function in dagit, add telemetry to gql requests

### PR DESCRIPTION
**Pull request by dpeng817**
Previously, we were using apollo hooks to log telemetry data. This doesn't actually match our use case very well, because we don't really care about the response to the telemetry log, and we want to log outside of the context of react components.

### CODE REVIEW

**Comment by bengotow** [comment]
hey hey! I don't have the context on this one but the code looks correct to me - we should probably rm the console.log and commented out bits (or put them back, looks like it might be a test :-) ) before we merge!

**Review by prha — COMMENTED** [review]
will defer to dish here

File: js_modules/dagit/packages/app/public/index.html
Comment (hellendag) [inline js_modules/dagit/packages/app/public/index.html:10]: No need to wrap this and key it on `initializationData` -- you can just add the new key to the existing object.

File: js_modules/dagit/packages/app/src/extractInitializationData.ts
Comment (hellendag) [inline js_modules/dagit/packages/app/src/extractInitializationData.ts:1]: Needs to be updated

File: js_modules/dagit/packages/app/src/index.tsx
Comment (hellendag) [inline js_modules/dagit/packages/app/src/index.tsx:22]: If telemetry is not enabled, we don't need to include `logTelemetryLink`.

File: js_modules/dagit/packages/core/src/app/Telemetry.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/Telemetry.tsx:11]: Within `@dagit/core` code, this should come from `AppContext` instead.

File: js_modules/dagit/packages/core/src/app/Telemetry.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/Telemetry.tsx:2]: `@dagit/app` may import from `@dagit/core`, but not vice versa. I should put a lint rule in place for that.

**Review by hellendag — CHANGES_REQUESTED** [review]

File: js_modules/dagit/packages/app/src/extractInitializationData.ts
Comment (hellendag) [inline js_modules/dagit/packages/app/src/extractInitializationData.ts:19]: eep

File: js_modules/dagit/packages/app/src/index.tsx
Comment (hellendag) [inline js_modules/dagit/packages/app/src/index.tsx:33]: Should be able to just do:

```
telemetryEnabled,
```

File: python_modules/dagit/dagit/app.py
Comment (hellendag) [inline python_modules/dagit/dagit/app.py:215]: I think this needs to replace the quotes as well, e.g. `"\"__TELEMETRY_ENABLED__\""`, otherwise you could end up with

```
"telemetry_action": "false",
```

File: python_modules/dagit/dagit_tests/test_app.py
Comment (hellendag) [inline python_modules/dagit/dagit_tests/test_app.py:97]: Also add an asset for the telemetry value?

**Review by hellendag — CHANGES_REQUESTED** [review]
Looks great! Requesting changes for the string replacement since I *think* that still needs the quotes to be replaced too.

File: python_modules/dagit/dagit/app.py
Comment (dpeng817) [reply python_modules/dagit/dagit/app.py:215]: I tested this in dagit and it seemed to work without the quotes.

File: js_modules/dagit/packages/app/src/extractInitializationData.ts
Comment (hellendag) [inline js_modules/dagit/packages/app/src/extractInitializationData.ts:23]: I guess this part of the conditional is unnecessary now, since it will have to be a boolean.

**Review by hellendag — APPROVED** [review]
🎉Thanks!

**Pull request merged**
